pkg/credentialprovider/default_provider.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package credentialprovider
+
+import (
+	"os"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+// A DockerConfigProvider that simply reads the .dockercfg file
+type defaultDockerConfigProvider struct{}
+
+// init registers our default provider, which simply reads the .dockercfg file.
+func init() {
+	RegisterCredentialProvider(".dockercfg",
```

Comment:
Wondering if this is suppose to replace something -- I thought we had a docker config provider already somewhere

Reply:
Ah I see the changes in pkg/credentialprovider/provider.go now

Reply:
What was the reason for this new file? Wondering if we should try to reduce the diff in this PR and leave it in provider.go?

Reply:
Agreed, I think it makes sense to merge it back.